test/src/main/java/org/apache/accumulo/test/functional/HalfClosedTabletIT.java

```diff
@@ -0,0 +1,279 @@
+  @Test
+  public void testBadIteratorOnStack() throws Exception {
+
+    // In this scenario the table is using an iterator for minc that is throwing an exception.
+    // This test ensures that the Tablet can still be unloaded normally by taking if offline
+    // after the bad iterator has been removed from the minc configuration.
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String tableName = getUniqueNames(1)[0];
+      final var tops = c.tableOperations();
+
+      tops.create(tableName);
+      final var tid = TableId.of(tops.tableIdMap().get(tableName));
+
+      IteratorSetting is = new IteratorSetting(30, BadIterator.class);
+      c.tableOperations().attachIterator(tableName, is, EnumSet.of(IteratorScope.minc));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation(new Text("r1"));
+        m.put("acf", tableName, "1");
+        bw.addMutation(m);
+      }
+
+      c.tableOperations().flush(tableName, null, null, false);
+
+      UtilWaitThread.sleepUninterruptibly(30, TimeUnit.SECONDS);
+
+      // minc should fail, so there should be no files
+      FunctionalTestUtils.checkRFiles(c, tableName, 1, 1, 0, 0);
+
+      // tell the server to take the table offline
+      tops.offline(tableName);
+
+      try {
+        // We are expecting this to fail, the table should not be taken offline
+        // because the bad iterator will prevent the minc from completing.
+        Wait.waitFor(() -> countHostedTablets(c, tid) == 0L, 120_000);
+        fail("Zero hosted tablets for table.");
+      } catch (IllegalStateException e) {}
+
+      // minc should fail, so there should be no files
+      FunctionalTestUtils.checkRFiles(c, tableName, 1, 1, 0, 0);
+
+      // remove the bad iterator. The failing minc task is in a backoff retry loop
+      // and should pick up this change on the next try
+      c.tableOperations().removeIterator(tableName, BadIterator.class.getSimpleName(),
+          EnumSet.of(IteratorScope.minc));
+
+      // Minc should have completed successfully
+      Wait.waitFor(() -> tabletHasExpectedRFiles(c, tableName, 1, 1, 1, 1), 340_000);
+
+      // Taking the table offline should succeed normally
+      tops.offline(tableName);
+      Wait.waitFor(() -> countHostedTablets(c, tid) == 0L, 340_000);
+    }
+  }
+
+  public static void setInvalidClassLoaderContextPropertyWithoutValidation(ServerContext context,
+      TableId tableId) {
+    TablePropKey key = TablePropKey.of(context, tableId);
+    context.getPropStore().putAll(key,
+        Map.of(Property.TABLE_CLASSLOADER_CONTEXT.getKey(), "invalid"));
+  }
+
+  public static void removeInvalidClassLoaderContextPropertyWithoutValidation(ServerContext context,
+      TableId tableId) {
+    TablePropKey key = TablePropKey.of(context, tableId);
+    context.getPropStore().removeProperties(key,
+        List.of(Property.TABLE_CLASSLOADER_CONTEXT.getKey()));
+  }
```

Comment:
I tried replacing this method locally w/ the following and changing the places where it was called and everything worked.

```suggestion
  public static void removeInvalidClassLoaderContextProperty(AccumuloClient client,
      String tableName) {
    try {
      client.tableOperations().removeProperty(tableName, Property.TABLE_CLASSLOADER_CONTEXT.getKey());
    } catch (AccumuloException | AccumuloSecurityException e) {
      throw new RuntimeException(e);
    }
  }
```

Reply:
Yeah, I guess we don't need to check the logs. If the property removal did not get picked up, then tests would not have passed. I'll apply.